lib/graphql/types/relay/base_connection.rb

```diff
@@ -90,10 +90,16 @@ def visible?(ctx)
 
           private
 
-          def define_nodes_field
-            field :nodes, [@node_type, null: true],
-              null: true,
-              description: "A list of nodes."
+          def define_nodes_field(nullable = true)
+            if nullable
+              field :nodes, [@node_type, null: true],
+                null: true,
+                description: "A list of nodes."
+            else
+              field :nodes, [@node_type],
+                null: false,
+                description: "A list of nodes."
+            end
```

Comment:
What do you think of DRYing this a bit? Eg,

```suggestion
          def define_nodes_field(nullable = true)
            field :nodes, [@node_type, null: null],
              null: null,
              description: "A list of nodes."
            end
```